src/document/doc-debug.js

```diff
@@ -64,7 +68,12 @@ function printDoc(doc) {
   }
 
   if (doc.type === "break-parent") {
-    return "breakParent";
+    const afterHardline =
+      Array.isArray(parentParts) &&
+      parentParts[index - 1] &&
+      parentParts[index - 1].type === "line" &&
+      parentParts[index - 1].hard;
+    return afterHardline ? undefined : "breakParent";
```

Comment:
Do you think it's easier to move this part to `flattenDoc`, add a function to replace `hardline + breakParent` as the new `parts`

https://github.com/thorn0/prettier/blob/2fad7569a9a9205ac222878bf21d5a53ff32cb50/src/document/doc-debug.js#L19
And
https://github.com/thorn0/prettier/blob/2fad7569a9a9205ac222878bf21d5a53ff32cb50/src/document/doc-debug.js#L37

Reply:
The current solution seems to be good enough. Also I'd like to keep it possible to easily comment out the `flattenDoc` call in order to see how the doc looks without it.

Reply:
https://github.com/prettier/prettier/pull/10169/files#diff-d5a6031cf2b293b180f03d67c8c402c7c13eb7358ccf04aa3cc4f1ad81f2fc4aR54

This part won't needed too.

We can replace `hardlineWithoutBreakParent + breakParent` with a new `type` while only use in this file, or mark the `breakParent` is after `hardlineWithoutBreakParent`, I think both okay, we won't need do the same check in two places.

Reply:
> mark the breakParent is after hardlineWithoutBreakParent

You mean in doc-builders? That didn't cross my mind. I'm okay with that.

Reply:
> keep it possible to easily comment out the flattenDoc call in order to see how the doc looks without it.

If that's what you need, we can mark the `breakParent` is after `hardline` when doing the check, so we don't need check twice.

Reply:
On the other hand, what if somebody in some plugin compares `breakParent` by reference? I'd rather keep things the way they are.

Reply:
Okay, good one.